flink-sql-runner/src/main/java/com/github/streamshub/flink/KubernetesSecretReplacer.java

```diff
@@ -0,0 +1,41 @@
+package com.github.streamshub.flink;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class KubernetesSecretReplacer {
+    private static final Pattern SECRET_PATTERN = Pattern.compile("\\{\\{secret:([^/]+)/([^}]+)/([^}]+)}}");
```

Comment:
a comment above showing a couple of real examples might help readers understand what the regex is trying to pull apart.